docs/pages/database-access/guides/clickhouse-self-hosted.mdx

```diff
@@ -0,0 +1,305 @@
+
+<ScopedBlock scope={["oss", "enterprise"]}>
+![Teleport Database Access Self-hosted ClickHouse](../../../img/database-access/guides/clickhouse_selfhosted_selfhosted.png)
+</ScopedBlock>
+<ScopedBlock scope={["cloud"]}>
+![Teleport Database Access ClickHouse Cloud](../../../img/database-access/guides/clickhouse_selfhosted_cloud.png)
+</ScopedBlock>
+
+## Prerequisites
+
+(!docs/pages/includes/commercial-prereqs-tabs.mdx!)
+
+- Either a Linux host or Kubernetes cluster where you will run the Teleport Database Service.
+
+You will also need the following, depending on the ClickHouse protocol you
+choose:
+
+<Tabs>
+<TabItem label="HTTP">
+- A self-hosted deployment of ClickHouse Server v22.3 or later.
+</TabItem>
+<TabItem label="Native TCP">
+- A self-hosted deployment of ClickHouse Server v23.3 or later.
+- The [clickhouse-client](https://clickhouse.com/docs/en/interfaces/cli) installed and added to your system's `PATH` environment variable.
+</TabItem>
+</Tabs>
+
+## Step 1/5. Create a Teleport token and user
+
+(!docs/pages/includes/database-access/token.mdx!)
+
+(!docs/pages/includes/database-access/create-user.mdx!)
+
+## Step 2/5. Create a certificate/key pair
+
+(!docs/pages/includes/database-access/tctl-auth-sign.mdx!)
+
+Follow the instructions below to generate TLS credentials for your database.
+
+```code
+# Export Teleport's certificate authority and a generated certificate/key pair
+# for host db.example.com with a 1-year validity period.
+$ tctl auth sign --format=db --host=clickhouse.example.com --out=server --ttl=8766h
+```
+
+(!docs/pages/includes/database-access/ttl-note.mdx!)
+
+The command will create three files:
+
+- `server.cas` with Teleport's certificate authority
+- `server.key` with a generated private key
+- `server.crt` with a generated host certificate
+
+## Step 3/5. Configure ClickHouse
+
+Use the generated secrets to enable mutual TLS in your `clickhouse-server/config.xml` configuration file:
+
+```xml
+<openSSL>
+    <server>
+       <privateKeyFile>/path/to/server.key</privateKeyFile>
+       <caConfig>/path/to/server.cas</caConfig>
+       <certificateFile>/path/to/server.crt</certificateFile>
+       <verificationMode>strict</verificationMode>
+    </server>
+</openSSL>
+```
+
+Additionally, your ClickHouse database user accounts must be configured to require a valid client certificate:
+
+```sql
+CREATE USER alice IDENTIFIED WITH ssl_certificate CN 'alice';
+```
+
+By default, the created user may not have access to anything and won't be able to connect, so let's grant it some permissions:
+
+```sql
+GRANT ALL ON *.* TO alice;
+```
+
+## Step 4/5. Configure and start the Database Service
+
+Install and configure Teleport on the host or Kubernetes cluster where you will
+run the Teleport Database Service:
+
+<Tabs>
+<TabItem label="Linux Server">
+
+(!docs/pages/includes/install-linux.mdx!)
+
+On the host where you will run the Teleport Database Service, start Teleport
+with the appropriate configuration.
```

Comment:
This sentence feels out of order, since it's not until L130 that you begin generating the appropriate configuration.